osmosis-apidb/src/main/java/org/openstreetmap/osmosis/apidb/v0_6/ApidbWriter.java

```diff
@@ -400,7 +404,7 @@ private void populateNodeParameters(PreparedStatement statement, int initialInde
 
         // We can't write an entity with a null timestamp.
         if (node.getTimestamp() == null) {
-            throw new OsmosisRuntimeException("Node " + node.getId() + " does not have a timestamp set.");
```

Comment:
Similar to an earlier comment, I'd prefer to see this entire line replaced with a call to an exception factory function rather than just centralising half of the message.